tests/validations/mta/mtaValidations.spec.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  Disposable,
+  mockFileSystemWatcher,
+  mockVscode,
+  testVscode,
+} from "../../mockUtil";
+mockVscode("src/validations/mta/mtaValidations");
+import * as sinon from "sinon";
+import { expect } from "chai";
+import {
+  DEV_MTA_EXT,
+  MTA_YAML,
+  updateMtaDiagnostics,
+  validateWsMtaYamls,
+  watchMtaYamlAndDevExtFiles,
+} from "../../../src/validations/mta/mtaValidations";
+import { Diagnostic, Range, Uri } from "vscode";
+import * as mtaDiagnostic from "../../../src/validations/mta/mtaDiagnostic";
+import { resolve } from "path";
+import * as fsExtra from "fs-extra";
+import { mta, Mta } from "@sap/mta-lib";
+
+describe("mtaValidations", () => {
+  const disposables: Disposable[] = [];
+
+  beforeEach(() => {
+    mtaDiagnostic.clearCurrentDiagnosticCollections();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("watchMtaYamlAndDevExtFiles", async () => {
+    it("registers to workspace.createFileSystemWatcher and to events with the correct parameters", async () => {
+      const createFileSystemWatcherSpy = sinon
+        .stub(testVscode.workspace, "createFileSystemWatcher")
+        .returns(mockFileSystemWatcher);
+      const onDidChangeSpy = sinon.stub(mockFileSystemWatcher, "onDidChange");
+      const onDidCreateSpy = sinon.stub(mockFileSystemWatcher, "onDidCreate");
+      const onDidDeleteSpy = sinon.stub(mockFileSystemWatcher, "onDidDelete");
+
+      watchMtaYamlAndDevExtFiles(disposables);
+
+      // createFileSystemWatcher
+      expect(createFileSystemWatcherSpy.callCount).to.equal(1);
+      expect(
+        createFileSystemWatcherSpy.withArgs(`**/{mta.yaml,dev.mtaext}`).called
+      ).to.be.true;
+
+      // onDidChange
+      expect(onDidChangeSpy.callCount).to.equal(1);
+      expect(
+        onDidChangeSpy.withArgs(sinon.match.func, undefined, disposables).called
+      ).to.be.true;
+
+      // onDidCreate
+      expect(onDidCreateSpy.callCount).to.equal(1);
+      expect(
+        onDidCreateSpy.withArgs(sinon.match.func, undefined, disposables).called
+      ).to.be.true;
+
+      // onDidDelete
+      expect(onDidDeleteSpy.callCount).to.equal(1);
+      expect(
+        onDidDeleteSpy.withArgs(sinon.match.func, undefined, disposables).called
+      ).to.be.true;
+    });
+
+    it("registers to workspace.onDidChangeWorkspaceFolders with the correct parameters", async () => {
```

Comment:
If you want, I think this description reflects the intention better:
```suggestion
    it("registers to workspace folder changes", async () => {
```